File: frontend/src/types/image.ts
export interface Image {
  image_id: string
  filename: string
  status: 'ready' | 'failed'
  file_size: number
  created_at: string
  updated_at: string
  error_message?: string
}

export interface ImageListResponse {
  images: Image[]
  total: number
}

export interface DeleteImageResponse {
  message: string
  image_id: string
}

export interface ImageChatMessage {
  role: 'user' | 'assistant'
  content: string
}

export interface ImageChatRequest {
  image_id: string
  message: string
  chat_history?: ImageChatMessage[]
}

export interface ImageChatResponse {
  response: string
}

export interface ImageChatHistoryResponse {
  image_id: string
  messages: ImageChatMessage[]
  total: number
}

File: backend/app/schemas/image.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from uuid import UUID


class ImageStatusResponse(BaseModel):
    image_id: UUID = Field(..., validation_alias="id")
    filename: str
    status: str
    file_size: int
    created_at: datetime
    updated_at: datetime
    error_message: Optional[str] = None

    class Config:
        from_attributes = True
        populate_by_name = True


class ImageListResponse(BaseModel):
    images: list[ImageStatusResponse]
    total: int


class DeleteImageResponse(BaseModel):
    message: str
    image_id: str


class ImageChatMessage(BaseModel):
    role: str
    content: str


class ImageChatRequest(BaseModel):
    image_id: str = Field(..., description="Image ID")
    message: str = Field(..., description="User question about the image")
    chat_history: Optional[list[ImageChatMessage]] = Field(default=[])


class ImageChatResponse(BaseModel):
    response: str


class ImageChatHistoryResponse(BaseModel):
    image_id: UUID
    messages: list[ImageChatMessage]
    total: int
